File: foliage-rpc/src/main/java/com/github/lavenderx/netty_rpc/protocol/RpcResponse.java
package com.github.lavenderx.netty_rpc.protocol;

import lombok.Data;

import java.io.Serializable;

@Data
public class RpcResponse implements Serializable {
    private static final long serialVersionUID = 1691010765111001804L;

    private String requestId;
    private String error;
    private Object result;

    public boolean isError() {
        return error != null;
    }
}

File: foliage-rpc/src/main/kotlin/org/lavenderx/foliage/nettyrpc/protocol/RpcResponse.kt
package org.lavenderx.foliage.nettyrpc.protocol

import java.io.Serializable

data class RpcResponse(val requestId: String) : Serializable {

    var error: String? = null
    var result: Any? = null

    val isError: Boolean = error == null

    companion object {
        private val serialVersionUID = 1691010765111001804L
    }
}